src/SfxWeb/src/app/Models/DataModels/Replica.ts

```diff
@@ -73,13 +73,31 @@ export class ReplicaOnPartition extends DataModelBase<IRawReplicaOnPartition> {
     }
 
     public get role(): string {
+        const previousRole: string = this.raw.PreviousReplicaRole ? `Previous Role: ${this.raw.PreviousReplicaRole} | ` : '';
+
+        if (this.parent.raw.PartitionStatus === 'Reconfiguring') {
+            return `Reconfiguring - ${previousRole}Target Role: ${this.raw.ReplicaRole}`;
+        }
+
+        return this.raw.ReplicaRole;
+    }
+
+    public get currentRole(): string {
```

Comment:
It appears that currentRole() is not used, could be removed since I want to change all reconfiguration texts from 'Reconfiguring - Target Role: X' to 'Reconfiguring - Previous Role: Y | Target Role: X'.